aptos-move/aptos-vm/src/block_executor/vm_wrapper.rs

```diff
@@ -62,16 +63,17 @@ impl<'a, S: 'a + StateView + Sync> ExecutorTask for AptosExecutorTask<'a, S> {
     // execution, or speculatively as a part of a parallel execution.
     fn execute_transaction(
         &self,
-        view: &impl StateView,
+        view: &(impl StateView + ResourceGroupResolver),
         txn: &PreprocessedTransaction,
         txn_idx: TxnIndex,
         materialize_deltas: bool,
     ) -> ExecutionStatus<AptosTransactionOutput, VMStatus> {
         let log_context = AdapterLogSchema::new(self.base_view.id(), txn_idx as usize);
+        let storage_adapter = &self.vm.as_move_resolver(view, Some((view, true)));
```

Comment:
I am wondering if we can avoid passing the same thing twice. The main issue I also have seen when adding support  for resource / module views: BlockSTM is essential an adapter (resolvers on top of state view), but we also have adapters which can be converted to view :/ 

What about adapter not implementing ResourceGroupResolver, but instead implementing it on a view level? This would also allow to fix respawned session adapter by dereferencing the adapter into that "view", adding change set on top, and re-wrapping into an adapter back, i.e.
```
1. Adapter (resources, modules (Move), configs, aggregators for native functions)
2. Fine-grained views including groups
3. StateView
```

The drawback is that we need two adapters though: one on top of state view and one on top of the other "views"